src/common.c

```diff
@@ -2951,30 +2951,81 @@ int stlink_erase_flash_page(stlink_t *sl, stm32_addr_t flashaddr) {
   return check_flash_error(sl);
 }
 
+// Check if an address and size are within the flash
+int stlink_check_address_range_validity(stlink_t *sl, stm32_addr_t addr, size_t size) {
+  if (addr < sl->flash_base || addr >= (sl->flash_base + sl->flash_size)) {
+    ELOG("Invalid address, it should be within 0x%08x - 0x%08lx\n", sl->flash_base, (sl->flash_base + sl->flash_size -1));
```

Comment:
Please, add explicit conversation to `unsigned long` for `(sl->flash_base + sl->flash_size -1)` .

```c
(unsigned long)(sl->flash_base + sl->flash_size -1)
```

Reply:
see comment below